# Astronomical Telescopes and Instruments 2020:
# Exercises on Spectrographs
# Due on 24 November at 14:15
# Auke Bruinsma, s1594443

## 1. Prism spectrograph
The MMTO telescope has a diameter of $6.5$m and a Cassegrain focal ratio of $f/15$. A spectrograph at the Cassegrain focus has an entrance slit set to match the typical seeing of $0.75$ arcseconds.

In [1]:
# Imports.
import numpy as np

In [2]:
# Setting up the given values.
D = 6.5 # Diameter [m].
N = 15 # Ratio.
phi = np.deg2rad(0.75/3600) # Seeing angle [rad].

**Question 1.1**: How wide is the slit in mm?

**Answer**:

Equations used:

$$ F = \frac{f}{D} \quad \text{slide 33, lecture 2} $$
    
$$ \phi = \frac{w}{f} \quad \text{slide 21, lecture 10} $$

Where $f$ is the focal length of the telescope and $w$ is the size of
the slit in mm. The angle is given in radians.

In [3]:
f = N * D # Focal length [m]
w = phi * f # Width of the slit [m]
print(f'Width of the slit: {w*1e3:.3f} mm')

Width of the slit: 0.355 mm


**Question 1.2**: The diameter of the collimator is $300$mm, and it projects onto a glass prism made of F2 glass with a base length $B = 200$mm. What is the dispersion $A$ in radians per nm at the Hydrogen alpha line? You
can assume $\frac{\text{d}n}{\text{d}\lambda} = 0.06 \mu\text{m}^{-1}$ at a wavelength of $0.656$ microns and the prism is set up in minimum
deviation.

**Answer**:

Dispersion of a glass prism:

$$ A = \frac{B}{d_\text{cam}} \frac{\text{d}n}{\text{d}\lambda} \quad \text{slide 26, lecture 10} $$

Because the prism is setup in minimum deviation, it may be assumed that $d_\text{cam} = d_\text{coll}$

In [4]:
d_coll = 300 # Diameter of the collimator [mm].
B = 200 # Base length F2 glass prism [mm].
dndl = 0.06 # \mu m^{-1}

A = B/d_coll * dndl*1e-3
print(f'The dispersion A at the Hydrogen alpha line: {A:.0e} rad/nm.')

The dispersion A at the Hydrogen alpha line: 4e-05 rad/nm.


**Question 1.3**: The focal ratio of the camera for the spectrograph is $f/2$. What is the slit width on the detector?

**Answer**:

$$ w' = rw \frac{f_\text{cam}}{f_\text{coll}} = r \phi f \frac{f_\text{cam}}{f_\text{coll}} \quad \text{slide 23, lecture 10} $$

$$ r = \frac{d_\text{coll}}{d_\text{cam}} \quad \text{slide 22, lecture 10} $$

$r \approx 1$, as was stated in the answer to question 1.2, so:

$$ w' = w \frac{f_\text{cam}}{f_\text{coll}} $$

$$ \frac{f}{D} = \frac{f_\text{coll}}{d_\text{coll}} \quad \text{slide 16, lecture 10} $$

Because the focal ratio of the camera equals $f/2$, and the Cassegrain focal ratio is $f/15$, the following computation gives the slit width on the detector

In [5]:
d_cam = d_coll # Diameter camera [mm].

w_accent = w * (2*d_cam)/(15*d_coll) * 1e6 # Slit width detector [μm]
print(f'Slit width on the detector: {w_accent:.3f} micrometer.')

Slit width on the detector: 47.269 micrometer.


**Question 1.4**: The detector is a CCD with $512$ pixels to a side and $20$ micron pixel size. Is the CCD adequately sampling the slit?

**Answer**:

Slide 48, lecture 10:

"The smallest resolution for the spectrograph should be sampled at the minimum of the Nyquist frequency, which is 2 pixels per resolution element."

The slit width on the detector was calculated in the previous exercise, and given is that each pixel is 20 microns in size. This gives the following ratio:

In [6]:
num_pixels = 512 # Number of pixels on a side for the CCD detector.
pixel_size = 20 # Pixel size [micron/pixel].

pixel_sample = w_accent / pixel_size # Number of pixels.
print(f'Number of pixels per resolution element: {pixel_sample:.2f} pixels')

Number of pixels per resolution element: 2.36 pixels


Since $2.36 > 2$, the answer is yes, the CCD is adequatly sampling the slit.

**Question 1.5**: What is the spectral resolution of the spectrograph at $656$nm?

**Answer**:

The spectral resolution is given in slide 2 of lecture 10:

$$ R = \frac{\lambda}{\Delta \lambda} $$

Information on the resolution element $\Delta \lambda$, from slide 18 of lecture 10:

"The resolution element is the minimum resolution of the spectrograph. This will depend of the spectral size of the image, which is a factor of image size, spectral magnification and the linear dispersion"

$$ \Delta \lambda = w\prime \frac{\text{d}\lambda}{\text{d}l} $$

The linear dispersion is given on slide 25 of lecture 10:

$$ \frac{\text{d}l}{\text{d}\lambda} = f_\text{cam} A $$

Combining the above formula, the spectra resolution can be found.

In [7]:
lam = 656 # Wavelength [nm].
f_cam = d_cam * 2 # Focal length camera [mm].
dldlam = f_cam * A * 1e6 # Linear dispersion [rad]
Deltalambda = w_accent / dldlam # Resoltuion element [μ/rad].
R = lam / Deltalambda / 1e3 # Spectral resolution.
print(f'Spectral resolution R at 656 nm: {R:.2f}')

Spectral resolution R at 656 nm: 333.07


**Question 1.6**: What is the estimated bandwidth of the spectrograph?

**Answer**: The bandwith can be estimated with the ratio of the size of CCD and the linear dispersion, as follows:

In [8]:
bandwidth = num_pixels * pixel_size / dldlam # Bandwith [μm]
print(f'Bandwidth: {bandwidth*1e3:.2f} nm')

Bandwidth: 426.67 nm


**Question 1.7**: Why can you only estimate the bandwidth?

**Answer**: Assumed was $\frac{\text{d}n}{\text{d}\lambda} = 0.06 \mu\text{m}^{-1}$ at a wavelength of $0.656$. This means the dispersion and the bandwidth is dependend on this value. But for other wavelengths $\frac{\text{d}n}{\text{d}\lambda}$ is not known. Therefore we can only estimate the bandwith for this specific wavelength. If we know the relation between $n$ and $\lambda$, it would be possible to give a better approximation. 

## 2. Diffraction grating
The James Webb Space Telescope has a diameter of $6.5$m and a focal length of $130$m.

In [9]:
# Initial data.
D = 6.5 # James Webb Space Telescope diameter [m].
f = 130 # James Webb Space Telescope focal length [m].

**Question 2.1**: What is the size of the Airy disk at the telescope focus for a wavelength of $2.2$ microns?

**Answer**: The size of the Airy disk at focus can be found by multiplying the diffraction limit $\theta = 1.22 \frac{\lambda}{D} $ with the focal length $f$:

In [10]:
lam = 2.2e-6 # Wavelength [m].
theta = 1.22 * lam / D # Diffraction limit.
size_airy_disk = theta * f # Size of the Airy disk [m].
print(f'Size of the Airy disk: {size_airy_disk*1e6:.3f} μm')

Size of the Airy disk: 53.680 μm


**Question 2.2**: What is the width of the spectrograph entrance slit that matches this diffraction limit?

**Answer**: The formula on slide 21 of lecture 10,

$$ \phi = \frac{w}{f} $$

states that the width is equal to the angle $\phi$ times the focal length $f$. So this is the same as the size of the Airy Disk, so $53.68$ μm. 

**Question 2.3**: Studying the dynamics of globular clusters requires a spectral resolution of $60000$. A diffraction grating with $100$ lines/mm is provided for the spectrograph. If the diffraction grating is used in 8th order in a Littrow configuration, what is the incident angle on the grating?

**Answer**:

Dispersion of diffraction grating formula from slide 31 of lecture 10:

$$ m \lambda = \sigma(\sin \alpha \pm \sin \beta) $$

$m$ is the order, $\sigma$ is the groove spacing (the number of mm between each ruled line) and $\alpha$ and $\beta$ are the angles.

From slide 47 of lecture 10, we know that $\alpha = \beta$ for a Littrow spectrograph. Then the equation above becomes:

$$ m \lambda = \sigma(\sin \alpha \pm \sin \alpha) $$
$$ m \lambda = 2 \sigma \sin \alpha $$

Which is also the equation that is found on slide 47 of lecture 10.
Rewriting gives:

$$ \alpha = \arcsin \frac{m \lambda}{2 \sigma} $$

Substituting all values gives.

In [11]:
spec_res = 6e4 # Spectral resolution.
dif_grat = 100 # Diffraction grating [lines/mm].
sigma = 1/dif_grat # Groove spacing [mm/line].
m = 8 # Order.

alpha = np.arcsin(m*lam/(2*(sigma*1e-3)))
print(f'Incident angle on the grating: {np.rad2deg(alpha):.2f} deg')

Incident angle on the grating: 61.64 deg


**Question 2.4**: What is the number of illuminated grooves needed for this spectral resolution?

**Answer**: Slide 35 of lecture 10 gives the spectral resolution:

$$ R = nm $$

where $m$ is the order and $n$ the total number of illumnated grooves. Subtituting the values gives:

In [12]:
n = spec_res / m # Number of illumnated grooves.
print(f'Number of illuminated grooves: {n:.0f}')

Number of illuminated grooves: 7500


**Question 2.5**: What is the full length of the grating?

**Answer**: There are 7500 illumnated grooves with a spacing of 100 lines per mm. This result in a length of $\frac{7500}{100} = 75$ mm for the grating.

In [13]:
l = n / dif_grat # Full length of the grating [mm].
print(f'Full length of the grating: {l:.0f} mm')

Full length of the grating: 75 mm


**Question 2.6**: What is the diameter of the collimated beam and its focal length?

**Answer**: Since that we know the length $l$ of the grating and in the incident angle $\alpha$ on the grating, the diameter of the beam can be found using geometry.

Next the focal length can be found using the formula on slide 15 of lecture 10.

$$ \frac{f}{D} = \frac{f_\text{coll}}{D_\text{coll}} $$

In [14]:
D_coll = l * np.cos(alpha) # Beam diameter collimated beam []
f_coll = f / D * D_coll # Focal length collimated beam.
print(f'Diameter collimated beam: {D_coll:.3f} mm')
print(f'Focal length collimated beam: {f_coll:.3f} mm')

Diameter collimated beam: 35.623 mm
Focal length collimated beam: 712.461 mm
